File: management-domain/Finance-department/Backend/Nodes/finance-automation-service/invoice-processing-service/src/shared/exceptions/conflict.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Exception thrown when a conflict occurs (e.g., duplicate resource)
 */
export class ConflictException extends BaseException {
  constructor(message: string, resource: string, context?: Record<string, unknown>) {
    super(
      message,
      HttpStatus.CONFLICT,
      'CONFLICT_ERROR',
      { resource, ...context },
    );
  }
}

File: shared-business-logics/services/universal-tracking-services/Backend/Java/universal-tracking-service/src/main/java/com/gogidix/shared/exceptions/ConflictException.java
package com.gogidix.shared.exceptions;

/**
 * Exception thrown when a conflict occurs (e.g., duplicate resource).
 */
public class ConflictException extends RuntimeException {

    public ConflictException(String message) {
        super(message);
    }

    public ConflictException(String message, Throwable cause) {
        super(message, cause);
    }
}
